### PR TITLE
feat: add tcp route data structures and cache

### PR DESCRIPTION
**Pull request by tokers**

### CODE REVIEW

File: pkg/apisix/nonexistentclient.go [hunk 1]
```diff
@@ -132,15 +160,19 @@ type dummyCache struct{}
...
+func (c *dummyCache) GetUpstream(_ string) (*v1.Upstream, error)       { return nil, cache.ErrNotFound }
+func (c *dummyCache) GetStreamRoute(_ string) (*v1.StreamRoute, error) { return nil, cache.ErrNotFound }
+func (c *dummyCache) ListRoutes() ([]*v1.Route, error)                 { return nil, nil }
+func (c *dummyCache) ListSSL() ([]*v1.Ssl, error)                      { return nil, nil }
+func (c *dummyCache) ListUpstreams() ([]*v1.Upstream, error)           { return nil, nil }
+func (c *dummyCache) ListStreamRoutes() ([]*v1.StreamRoute, error)     { return nil, nil }
+func (c *dummyCache) DeleteRoute(_ *v1.Route) error                    { return nil }
+func (c *dummyCache) DeleteSSL(_ *v1.Ssl) error                        { return nil }
+func (c *dummyCache) DeleteUpstream(_ *v1.Upstream) error              { return nil }
+func (c *dummyCache) DeleteStreamRoute(_ *v1.StreamRoute) error        { return nil }
```
Comment (gxthrj) [inline pkg/apisix/nonexistentclient.go:178]: Can these methods be refactored to be together, 
such as, dummyCache.Insert(), dummyCache.List()

File: pkg/apisix/stream_route.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+func (r *streamRouteClient) Update(ctx context.Context, obj *v1.StreamRoute) (*v1.StreamRoute, error) {
+	log.Debugw("try to update stream_route",
+		zap.String("id", obj.ID),
+		zap.String("cluster", "default"),
+		zap.String("url", r.url),
+	)
+	if err := r.cluster.HasSynced(ctx); err != nil {
+		return nil, err
+	}
+	// FIXME use unified v1.Route, removing routeReqBody.
```
Comment (gxthrj) [inline pkg/apisix/stream_route.go:203]: Confusing comments.

**Comment by gxthrj** [comment]
BTW, There is no e2e test for stream route.

File: pkg/apisix/nonexistentclient.go (hunk 1 above)
Comment (tokers) [reply pkg/apisix/nonexistentclient.go:178]: OK, will change it.

**Comment by tokers** [comment]
@gxthrj This PR just add the data structure for it.

**Comment by tokers** [comment]
@gxthrj Fixed.

**Review by gxthrj — APPROVED** [review]

**Pull request merged**